protocol/messenger.go

```diff
@@ -1307,7 +1304,7 @@ func (m *Messenger) SendChatMessage(ctx context.Context, message *Message) (*Mes
 		return nil, errors.New("Chat not found")
 	}
 
-	err := extendMessageFromChat(message, chat, &m.identity.PublicKey)
+	err := extendMessageFromChat(message, chat, &m.identity.PublicKey, m.transport)
```

Comment:
I would recommend to have a getter on `Messenger`:
```suggestion
	err := extendMessageFromChat(message, chat, &m.identity.PublicKey, m.getTimeSource())
```
so that when the implementation changes, you will only need to change it in one place, `getTimeSource` method. It looks odd that we pass `transport` to `extendMessageFromChat` which is a helper.